test/io_streams_test.py

```diff
@@ -0,0 +1,43 @@
+# Copyright Modal Labs 2024
+from modal import enable_output
+from modal.io_streams import StreamReader
+from modal_proto import api_pb2
+
+
+def test_stream_reader(servicer, client):
+    lines = ["foo\n", "bar\n", "baz\n"]
+
+    async def sandbox_get_logs(servicer, stream):
+        await stream.recv_message()
+
+        # Data with multiple lines
+        log1 = api_pb2.TaskLogs(
+            data="".join(lines),
+            file_descriptor=api_pb2.FILE_DESCRIPTOR_STDOUT,
+        )
+        await stream.send_message(api_pb2.TaskLogsBatch(entry_id="0", items=[log1]))
+
+        # Data with single lines sent separately
+        for i, line in enumerate(lines):
+            log = api_pb2.TaskLogs(data=line, file_descriptor=api_pb2.FILE_DESCRIPTOR_STDOUT)
+            await stream.send_message(api_pb2.TaskLogsBatch(entry_id=str(i + 1), items=[log]))
+
+        # Send EOF
+        await stream.send_message(api_pb2.TaskLogsBatch(eof=True))
+
+    with servicer.intercept() as ctx:
```

Comment:
Can you break these into separate tests and add comments? It's a bit hard to understand what's going on. Also add a test that partial lines are aggregated.